test/integration/test-amp-ad-3p.js

```diff
@@ -187,15 +187,46 @@ describe('amp-ad 3P', () => {
           5000
         );
       })
-      .then(() => {
+      .then(async function () {
         lastIO = null;
         iframe.contentWindow.context.observeIntersection((changes) => {
           lastIO = changes[changes.length - 1];
         });
+        await poll('wait for initial IO entry', () => {
+          return (
+            lastIO != null &&
+            lastIO.boundingClientRect.top == 1000 &&
+            lastIO.intersectionRatio == 1
+          );
+        });
+        lastIO = null;
         fixture.win.scrollTo(0, 1000);
-        fixture.win.document.body.dispatchEvent(new Event('scroll'));
-        return poll('wait for new IO entry', () => {
-          return lastIO != null;
+        fixture.win.dispatchEvent(new Event('scroll'));
+        await poll('wait for new IO entry when ad is fully visible', () => {
+          return (
+            lastIO != null &&
+            lastIO.boundingClientRect.top == (platform.isIos() ? 1 : 0) &&
+            lastIO.intersectionRatio == 1
+          );
+        });
+        lastIO = null;
+        fixture.win.scrollTo(0, 1125);
```

Comment:
might explain the magic number 1125.

also, one thing to be cautious, what's the window size?

Reply:
The iframe size is 3000px. I added the 5000px div to the document to make it scrollable. 

The ad is placed at 1000px, with height 250px. This is to make the ad partially visible.